data_structures/src/chain/priority.rs

```diff
@@ -0,0 +1,1169 @@
+
+impl Visitor for PriorityVisitor {
+    type Visitable = (Transaction, /* fee */ u64, /* weight */ u32);
+
+    fn visit(&mut self, (transaction, fee, weight): &Self::Visitable) {
+        match transaction {
+            Transaction::DataRequest(_) => {
+                self.0
+                    .digest_drt_priority(Priority::from_fee_weight(*fee, *weight));
+            }
+            Transaction::ValueTransfer(_) => {
+                self.0
+                    .digest_vtt_priority(Priority::from_fee_weight(*fee, *weight));
+            }
+            _ => (),
+        }
+    }
+}
+
+impl StatefulVisitor for PriorityVisitor {
+    type State = Priorities;
+
+    fn take_state(self) -> Self::State {
+        self.0
+    }
+}
+
+/// A whole set of estimates for priority of DRT and VTT transactions.
+///
+/// Each estimate contains values for 5 different tiers of priority:
+/// - Stinky
+/// - low
+/// - Medium
+/// - High
+/// - Opulent
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct PrioritiesEstimate {
+    pub drt_stinky: PriorityEstimate,
+    pub drt_low: PriorityEstimate,
+    pub drt_medium: PriorityEstimate,
+    pub drt_high: PriorityEstimate,
+    pub drt_opulent: PriorityEstimate,
+    pub vtt_stinky: PriorityEstimate,
+    pub vtt_low: PriorityEstimate,
+    pub vtt_medium: PriorityEstimate,
+    pub vtt_high: PriorityEstimate,
+    pub vtt_opulent: PriorityEstimate,
+}
+
+impl PrioritiesEstimate {
+    /// Show a nicely formatted table with the priority and time-to-block estimates for different
+    /// priority tiers.
+    ///
+    /// The `time_formatter` function allows to define how to format the time-to-block.
+    #[allow(clippy::to_string_in_format_args)]
+    pub fn pretty_print<F>(&self, mut time_formatter: F) -> String
+    where
+        F: FnMut(&TimeToBlock) -> String,
+    {
+        format!(
+            r#"╔══════════════════════════════════════════════════════════╗
+║ TRANSACTION PRIORITY ESTIMATION REPORT                   ║
+╠══════════════════════════════════════════════════════════╣
+║ Data request transactions                                ║
+╟──────────┬──────────┬────────────────────────────────────║
+║     Tier │ Priority │ Time-to-block                      ║
+╟──────────┼──────────┼────────────────────────────────────║
+║   Stinky │ {:<8} │ {:<33}  ║
+║      Low │ {:<8} │ {:<33}  ║
+║   Medium │ {:<8} │ {:<33}  ║
+║     High │ {:<8} │ {:<33}  ║
+║  Opulent │ {:<8} │ {:<33}  ║
+╠══════════════════════════════════════════════════════════╣
+║ Value transfer transactions                              ║
+╟──────────┬──────────┬────────────────────────────────────║
+║     Tier │ Priority │ Time-to-block                      ║
+╟──────────┼──────────┼────────────────────────────────────║
+║   Stinky │ {:<8} │ {:<33}  ║
+║      Low │ {:<8} │ {:<33}  ║
+║   Medium │ {:<8} │ {:<33}  ║
+║     High │ {:<8} │ {:<33}  ║
+║  Opulent │ {:<8} │ {:<33}  ║
```

Comment:
The priority column should be right-aligned:

```suggestion
║  Opulent │ {:>8} │ {:<33}  ║
```

```
   0.000
 578.817
 680.962
 783.106
1150.000
```

Instead of 

```
0.000   
577.913 
679.897 
781.882 
1150.000
```